orchestrator/api/helpers.py

```diff
@@ -73,86 +73,82 @@ def _add_sort_to_query(query: Query, sort: Optional[List[str]]) -> Query:
+                        stmt = stmt.filter(func.lower(ProductTable.tag).in_([s.lower() for s in sub_values]))
                     elif field == "product":
                         sub_values = value.split("-")
-                        query = query.filter(func.lower(ProductTable.name).in_([s.lower() for s in sub_values]))
+                        stmt = stmt.filter(func.lower(ProductTable.name).in_([s.lower() for s in sub_values]))
                     elif field == "status":
                         # For React table 7
                         statuses = value.split("-")
-                        query = query.filter(SubscriptionTable.status.in_([s.lower() for s in statuses]))
+                        stmt = stmt.filter(SubscriptionTable.status.in_([s.lower() for s in statuses]))
                     elif field == "statuses":
                         # For port subscriptions
                         sub_values = value.split("-")
-                        query = query.filter(SubscriptionTable.status.in_([s.lower() for s in sub_values]))
+                        stmt = stmt.filter(SubscriptionTable.status.in_([s.lower() for s in sub_values]))
                     elif field == "organisation":
-                        query = query.filter(SubscriptionTable.customer_id == value)
+                        stmt = stmt.filter(SubscriptionTable.customer_id == value)
                     elif field == "tsv":
                         # Quote key:value tokens. This will use the FOLLOWED BY operator (https://www.postgresql.org/docs/13/textsearch-controls.html)
                         processed_text_query = _process_text_query(value)
 
                         logger.debug("Running full-text search query:", value=processed_text_query)
                         # TODO: Make 'websearch_to_tsquery' into a sqlalchemy extension
-                        query = query.join(SubscriptionSearchView).filter(
+                        stmt = stmt.join(SubscriptionSearchView).filter(
                             func.websearch_to_tsquery("simple", processed_text_query).op("@@")(
                                 SubscriptionSearchView.tsv
                             )
                         )
 
                     elif field in SubscriptionTable.__dict__:
-                        query = query.filter(cast(SubscriptionTable.__dict__[field], String).ilike("%" + value + "%"))
+                        stmt = stmt.filter(cast(SubscriptionTable.__dict__[field], String).ilike("%" + value + "%"))
 
-    query = _add_sort_to_query(query, sort)
+    return _add_sort_to_query(stmt, sort)
 
-    if range is not None and len(range) == 2:
+
+def add_response_range(stmt: Selectable, range_: Optional[list[int]], response: Response) -> Selectable:
+    if range_ is not None and len(range_) == 2:
+        total = db.session.scalar(select(func.count()).select_from(stmt.subquery()))
+        range_start = int(range_[0])
```

Comment:
Is this cast to int needed, since it's a list of int?

If not, you could do:

range_start, range_end = range_